pkg/quota/v1/evaluator/core/persistent_volume_claims.go

```diff
@@ -160,17 +160,42 @@ func (p *pvcEvaluator) Usage(item runtime.Object) (corev1.ResourceList, error) {
 	}
 
 	// charge for storage
-	if request, found := pvc.Spec.Resources.Requests[corev1.ResourceStorage]; found {
-		result[corev1.ResourceRequestsStorage] = request
+	requestedStorage := p.getStorageUsage(pvc)
+	if requestedStorage != nil {
+		result[corev1.ResourceRequestsStorage] = *requestedStorage
 		// charge usage to the storage class (if present)
 		if len(storageClassRef) > 0 {
 			storageClassStorage := corev1.ResourceName(storageClassRef + storageClassSuffix + string(corev1.ResourceRequestsStorage))
-			result[storageClassStorage] = request
+			result[storageClassStorage] = *requestedStorage
 		}
 	}
+
 	return result, nil
 }
 
+func (p *pvcEvaluator) getStorageUsage(pvc *corev1.PersistentVolumeClaim) *resource.Quantity {
```

Comment:
nit: By scoping down method parameters to the bare minimum required it makes code easier to test and more robust (since each method only needs to validate the input it cares about) - i.e. input here could be `resourcesRequests` instead of the entire `PVC` object.

Reply:
that is faire but in this case we need both `pvc.Spec.AllocatedResources` and `pvc.Spec.Resources` , so will end up passing both fields individually or pass full PVC object.

Reply:
I would opt to pass both fields in that case. But that's a personal preference.